kaniko/code/main.go

```diff
@@ -115,17 +121,52 @@ func main() {
 	// Launch a timer to measure the time needed to parse/copy/extract
 	start := time.Now()
 
-	// Build the Dockerfile
-	logrus.Infof("Building the %s", b.DockerFileName)
-	err := b.BuildDockerFile()
-	if err != nil {
-		panic(err)
+	logrus.Info("Finding the Dockerfiles to build ...")
+	metadatafileNameToParse := os.Getenv("METADATA_FILE_NAME")
+	logrus.Infof("METADATA TOML FILE: %s", metadatafileNameToParse)
+	if metadatafileNameToParse == "" {
+		metadatafileNameToParse = defaultMetadataFileName
 	}
-	err = reapChildProcesses()
+
+	logrus.Info("Finding base image to provide ...")
+	baseimage := os.Getenv("CNB_BASE_IMAGE")
+	logrus.Infof("CNB_BASE_IMAGE: %s", baseimage)
+
+	// Read metadata toml
+	meta := model.Metadata{}
+	_, err := toml.DecodeFile(metadatafileNameToParse, &meta)
 	if err != nil {
 		panic(err)
 	}
 
+	toMultiArg := func(args []model.DockerfileArg) []string {
+		var result []string
+		for _, arg := range args {
+			result = append(result, fmt.Sprintf("%s=%s", arg.Key, arg.Value))
+		}
+
+		result = append(result, fmt.Sprintf(`base_image=%s`, baseimage))
+		return result
+	}
+
+	for _, d := range meta.Dockerfiles {
+		// Define opts
+		opts := b.Opts
+		opts.DockerfilePath = d.Path
+		opts.BuildArgs = toMultiArg(d.Args)
```

Comment:
So finally we will only parse the Build atgs and not the run args here as I see that you changed the model ?